src/common/fanctl.h

```diff
@@ -0,0 +1,123 @@
+// fanctl.h
+#ifndef _FANCTL_H
+#define _FANCTL_H
+
+#include <inttypes.h>
+#include <stdbool.h>
+
+#define FANCTL_MAX_FANS 2 // maximum number of fans for C wrapper functions
+
+// this structure contain variables for software pwm fan control with phase-shifting
+typedef struct _fanctl_pwm_t {
+    uint8_t pin : 8; // output pwm pin
+    uint8_t min : 8; // minimum pwm value
+    uint8_t max : 8; // maximum pwm value
+    union {
+        struct {              // flags:
+            bool ini : 1;     //  hw initialized
+            bool out : 1;     //  current pwm output state (0/1)
+            bool pha_ena : 1; //  phase shift enabled
```

Comment:
make sure you don't rely on the ordering of bits - the compiler is allowed to reorder the bits arbitrarily

Reply:
I want just save RAM, maybe we can remove the union/flg and keep only the structure with bit addressed members.